### PR TITLE
GitHub API: handles comments

### PR DESCRIPTION
**Pull request by Morriar**
This time for comments:

* commit comments
* issue comments
* diff comments

### CODE REVIEW

File: lib/github/api.nit [hunk 1]
```diff
@@ -267,6 +267,60 @@ class GithubAPI
...
+	#     assert repo != null
+	#     var comment = api.load_review_comment(repo, 21010363)
+	#     assert comment.path == "src/modelize/modelize_property.nit"
+	#     assert comment.original_position == 26
+	#     assert comment.pull.number == 945
+	fun load_review_comment(repo: Repo, id: Int): nullable ReviewComment do
+		var comment = new ReviewComment(self, repo, id)
+		comment.load_from_github
+		if was_error then return null
+		return comment
```
Comment (privat) [inline lib/github/api.nit]: Je me demande si les trois

~~~nit
comment.load_from_github
if was_error then return null
return comment
~~~

pourraient pas etre factorisés.

File: lib/github/api.nit [hunk 2]
```diff
@@ -631,6 +685,29 @@ class Issue
...
 
+	# List of comments made on this issue.
+	fun comments: Array[IssueComment] do
+		var res = new Array[IssueComment]
+		var count = comments_count
+		var page = 1
+		var array = api.get("{key}/comments?page={page}")
+		if not array isa JsonArray then
+			print "error"
+			print api.last_error.to_s
```
Comment (privat) [inline lib/github/api.nit]: c'est pas `last_error` qui gere les erreur, c'est pas a la lib de faire des prints.

File: lib/github/api.nit [hunk 3]
```diff
@@ -848,3 +925,129 @@ class Milestone
...
 	end
 end
+
+# A Github comment
+#
+# There is two kinds of comments:
+#
+# * `IssueComment` are made on an issue or pull request page.
+# * `DiffComment` are made on the diff associated to a pull request.
+abstract class Comment
```
Comment (privat) [inline lib/github/api.nit]: ya pas aussi les commentaires associés aux commits ?

**Comment by privat** [comment]
+1 mais je me demand, pourquoi `to_s.to_i` plutot que directement `to_i` ?

File: lib/github/api.nit (hunk 1 above)
Comment (Morriar) [inline lib/github/api.nit]: How do you see such a factorization?

~~~
private fun do_comment_loading(comment: Comment): nullable Comment do
    comment.load_from_github
    if was_error then return null
    return comment
end
~~~

Then in the code at the 3 call sites:

~~~
var comment = new XComment(x, y, z)
if do_comment_loading(comment) == null then return null else return comment
~~~

**Comment by Morriar** [comment]
Like in `json["chose"].to_s.to_i`?

Because `json[]` returns a `nullable Jsonable`, `to_s` is used to "cast" it as a String then call the `to_i`.

File: lib/github/api.nit (hunk 1 above)
Comment (privat) [inline lib/github/api.nit]: `return comment.load_from_github` ?

**Comment by R4PaSs** [comment]
+1

**Comment by Morriar** [comment]
Reroll:

* replaces all remaining inverted assertion with `assert x != null`
* factorizes `load_from_github`
* removes remaining debugging prints

**Pull request merged**